src/app/pages/checkout-payment/checkout-payment/checkout-payment.component.html

```diff
@@ -1,5 +1,5 @@
 <div class="row" *ngIf="basket">
-  <div class="col-md-12">
+  <div class="col-md-12" [ishScroll]="basketHasError" [scrollContainer]="documentElement" [scrollDuration]="500">
```

Comment:
I think the current way you access the `documentElement` is not very pretty. I think it would be a lot cleaner if the component didn't have to know about any DOM logic but instead only takes care of providing data and form logic. My suggestions:
-  Get the `documentElement` in the directive from the target element by going up the tree and finding the root element. You can control this behavior with an Input flag (for example, `@Input() scrollContainer: 'parent' | 'root' | HTMLElement`) and then decide in the directive based on that.
- Definitely don't require the component to get some part of the DOM. It's quite "un-angular" and there are better ways. For example, if you want to provide a specific `Element` that should scroll, you can use a [template reference variable](https://angular.io/guide/template-reference-variables#template-variable-scope) like this: 

```html
<div class="row" *ngIf="basket" #example>
  <div class="col-md-12" [ishScroll]="basketHasError" [scrollContainer]="example" [scrollDuration]="500">
  </div>
</div>
```

Of course you have to take care to see what kind of type `example` will be (I think `ElementRef`) but this would enable you to skip the component and setup this logic directly in the template.